File: iks-core/src/com/iks/cms/core/sql/MatchCriteria.java
package com.iks.cms.core.sql;

import com.iks.cms.core.query.*;

import org.apache.commons.lang3.*;

/**
 * @author Igor Kaynov
 */
public class MatchCriteria implements ICriteria {
  private Column column;
  private String condition;
  private Object value;
  public MatchCriteria( Column column, Object value ) {
    this(column, value, MatchType.Eq);
  }
  public MatchCriteria( Column column, Object value, MatchType matchType ) {
    this.column = column;
    this.value = value;
    this.condition = matchType.getCondition();
  }
  public String toString() {
    String valueStr = value == null ? null : StringUtils.trimToNull( value.toString() );
    if (valueStr == null) {
      return column.toString() + " is null";
    } else {
      return column.toString() + condition + "'" + value + "'";
    }
  }
}

File: iks-core/src/com/iks/cms/core/sql/MatchCriteria.kt
package com.iks.cms.core.sql

import com.iks.cms.core.query.*

import org.apache.commons.lang3.*

/**
 * @author Igor Kaynov
 */
class MatchCriteria @JvmOverloads constructor(private val column: Column, private val value: Any?, matchType: MatchType = MatchType.Eq) : ICriteria {
  private val condition: String

  init {
    this.condition = matchType.condition
  }

  override fun toString(): String {
    val valueStr = if (value == null) null else StringUtils.trimToNull(value.toString())
    if (valueStr == null) {
      return column.toString() + " is null"
    } else {
      return column.toString() + condition + "'" + value + "'"
    }
  }
}
